src/pipeline/src/mng/table.rs

```diff
@@ -0,0 +1,358 @@
+            PIPELINE_TABLE_NAME,
+            table_info.full_table_name(),
+            output
+        );
+
+        Ok(())
+    }
+
+    pub async fn get_pipeline(
+        &self,
+        schema: &str,
+        name: &str,
+    ) -> Result<Pipeline<GreptimeTransformer>> {
+        if let Some(pipeline) = self.get_compiled_pipeline_from_cache(schema, name) {
+            return Ok(pipeline);
+        }
+
+        let pipeline = self.find_pipeline_by_name(schema, name).await?;
+        let compiled_pipeline = Self::compile_pipeline(&pipeline)?;
+        self.pipelines.write().unwrap().insert(
+            Self::generate_pipeline_cache_key(schema, name),
+            compiled_pipeline.clone(),
+        );
+        Ok(compiled_pipeline)
+    }
+
+    pub async fn insert_and_compile(
+        &self,
+        schema: &str,
+        name: &str,
+        content_type: &str,
+        pipeline: &str,
+    ) -> Result<Pipeline<GreptimeTransformer>> {
+        let compiled_pipeline = Self::compile_pipeline(pipeline)?;
+
+        self.insert_pipeline_to_pipeline_table(schema, name, content_type, pipeline)
+            .await?;
+
+        self.pipelines.write().unwrap().insert(
+            Self::generate_pipeline_cache_key(schema, name),
+            compiled_pipeline.clone(),
+        );
+
+        Ok(compiled_pipeline)
+    }
+
+    async fn find_pipeline_by_name(&self, schema: &str, name: &str) -> Result<String> {
+        let table_info = self.table.table_info();
+
+        let table_name = TableReference::full(
+            table_info.catalog_name.clone(),
+            table_info.schema_name.clone(),
+            table_info.name.clone(),
+        );
+
+        let table_provider = Arc::new(DfTableProviderAdapter::new(self.table.clone()));
+        let table_source = Arc::new(DefaultTableSource::new(table_provider));
+
+        let plan = LogicalPlanBuilder::scan(table_name, table_source, None)
+            .context(BuildDfLogicalPlanSnafu)?
+            .filter(and(
+                col("schema").eq(lit(schema)),
+                col("name").eq(lit(name)),
+            ))
+            .context(BuildDfLogicalPlanSnafu)?
+            .project(vec![col("pipeline")])
+            .context(BuildDfLogicalPlanSnafu)?
+            .build()
+            .context(BuildDfLogicalPlanSnafu)?;
+
+        let output = self
+            .query_engine
+            .execute(LogicalPlan::DfPlan(plan), Self::query_ctx(&table_info))
+            .await
+            .context(ExecuteInternalStatementSnafu)?;
+        let stream = match output.data {
+            OutputData::Stream(stream) => stream,
+            OutputData::RecordBatches(record_batches) => record_batches.as_stream(),
+            _ => unreachable!(),
+        };
+
+        let records = record_util::collect(stream)
+            .await
+            .context(CollectRecordsSnafu)?;
+
+        ensure!(!records.is_empty(), PipelineNotFoundSnafu { name });
+        // assume schema + name is unique for now
```

Comment:
It's not a good idea to assume it. It's better to set the primary keys to be `(schema, name)`.

Reply:
primary keys is (schema,name). but ts field will be different.
So I added `order by created_at desc limit 1` query condition.